planner/core/exhaust_physical_plans.go

```diff
@@ -1263,10 +1263,43 @@ func (la *LogicalAggregation) getHashAggs(prop *property.PhysicalProperty) []Phy
 	return hashAggs
 }
 
+func copyProperties(props []*property.PhysicalProperty) []*property.PhysicalProperty {
+	retProp := make([]*property.PhysicalProperty, 0, len(props))
+	for _, prop := range props {
+		temp := *prop
+		retProp = append(retProp, &temp)
+	}
+	return retProp
+}
+
+func (ps *PhysicalStreamAgg) convert2Enforced() {
+	newProps := copyProperties(ps.childrenReqProps)
+	for i := range newProps {
+		newProps[i].Enforced = true
+	}
+	ps.childrenReqProps = newProps
+}
+
 func (la *LogicalAggregation) exhaustPhysicalPlans(prop *property.PhysicalProperty) []PhysicalPlan {
+	preferHash := (la.preferAggType & preferHashAgg) > 0
+	preferStream := (la.preferAggType & preferStreamAgg) > 0
+
+	hashAggs := la.getHashAggs(prop)
+	if hashAggs != nil && preferHash && !preferStream {
```

Comment:
If `preferHash` is true, then `preferStream` must be false since there is at most one hint?

Reply:
Yes, you are right. This is for further assurance, if we don't want to throw an error when there are more than one hints.